### PR TITLE
Fix widgets with built-in scrollbars

### LINKED ISSUE #83819: Widgets with built-in scrollbars that need to account for ScrollBehavior
Noticed in https://github.com/flutter/flutter/pull/81278 that both the `CupertinoActionSheet` and `CupertinoAlertDialog` have built-in scrollbars. This would cause an issue for scrollbars applied by the inherited ScrollBehavior. These widgets just need to be updated to account for this.

### PR DESCRIPTION
**Pull request by Piinks**
Fixes #83819 

This updates widgets that have built-in scrollbars to not have additional scrollbars applied by inherited ScrollBehaviors.

Affected widgets are:
- DropdownMenuButton
- CupertinoActionSheet
- CupertinoAlertDialog
- LicensePage

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/about_test.dart [hunk 1]
```diff
@@ -705,6 +706,50 @@ void main() {
...
+      case TargetPlatform.iOS:
+        expect(find.byType(CupertinoScrollbar), findsOneWidget);
+        break;
+      case null:
+        break;
+    }
+    expect(find.byType(Scrollbar), findsOneWidget);
+    expect(find.byType(RawScrollbar), findsNothing);
+
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{ ...TargetPlatform.values }));
```
Comment (gspencergoog) [inline packages/flutter/test/material/about_test.dart:752]:
```suggestion
  }, variant: const TargetPlatformVariant.all());
```

File: packages/flutter/test/material/dropdown_test.dart [hunk 2]
```diff
@@ -3449,4 +3450,26 @@ void main() {
...
+    await tester.pump();
+    await tester.pumpAndSettle(); // finish the menu animation
+
+    // The inherited ScrollBehavior should not apply Scrollbars since they are
+    // already built in to the widget.
+    expect(find.byType(CupertinoScrollbar), findsNothing);
+    expect(find.byType(Scrollbar), findsOneWidget);
+    expect(find.byType(RawScrollbar), findsNothing);
+
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{ ...TargetPlatform.values }));
```
Comment (gspencergoog) [inline packages/flutter/test/material/dropdown_test.dart:3474]:
```suggestion
  }, variant: const TargetPlatformVariant.all());
```

File: packages/flutter/test/cupertino/dialog_test.dart [hunk 3]
```diff
@@ -1247,6 +1248,45 @@ void main() {
...
+    await tester.pump();
+
+    // The inherited ScrollBehavior should not apply Scrollbars since they are
+    // already built in to the widget.
+    expect(find.byType(Scrollbar), findsNothing);
+    expect(find.byType(RawScrollbar), findsNothing);
+    // Built in CupertinoScrollbars should only number 2: one for the actions,
+    // one for the content.
+    expect(find.byType(CupertinoScrollbar), findsNWidgets(2));
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{ ...TargetPlatform.values }));
```
Comment (gspencergoog) [inline packages/flutter/test/cupertino/dialog_test.dart:1289]:
```suggestion
  }, variant: const TargetPlatformVariant.all());
```

File: packages/flutter/test/cupertino/action_sheet_test.dart [hunk 4]
```diff
@@ -993,6 +994,46 @@ void main() {
...
+    await tester.pump();
+
+    // The inherited ScrollBehavior should not apply Scrollbars since they are
+    // already built in to the widget.
+    expect(find.byType(Scrollbar), findsNothing);
+    expect(find.byType(RawScrollbar), findsNothing);
+    // Built in CupertinoScrollbars should only number 2: one for the actions,
+    // one for the content.
+    expect(find.byType(CupertinoScrollbar), findsNWidgets(2));
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{ ...TargetPlatform.values }));
```
Comment (gspencergoog) [inline packages/flutter/test/cupertino/action_sheet_test.dart:1036]:
```suggestion
  }, variant: const TargetPlatformVariant.all());
```

**Review by gspencergoog — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/about_test.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/test/material/about_test.dart:752]: Oh goodness! My brain must've still been on vacation. 😋  Thanks!

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**